django_cloud_deploy/cli/prompt.py

```diff
@@ -32,260 +33,430 @@
-        project_name = project_client.get_project(project_id)['name']
-        message = 'Project name found: {}'.format(project_name)
-        console.tell(('{} {}').format(step_prompt, message))
-        return project_name
+        pass
 
-    @staticmethod
-    def validate(s, credentials: Optional[credentials.Credentials] = None):
-        """Validates that a string is a valid project name.
+    @abc.abstractmethod
+    def _is_valid_passed_arg(self, console: io.IO, step: str,
+                             value: Optional[str],
+                             validate: Callable[[str], None]) -> bool:
+        """Used to validate if the user passed in a parameter as a flag.
 
-        Args:
-            s: The string to validate.
-            credentials: The OAuth2 Credentials object to use for api calls
-                during validation.
+        All s that retrieve a parameter should call this function first.
+        It requires all s to have implemented validate. The code also
+        will process a passed in paramater as a step. This is used to have a
+        hard coded amount of steps that is easy to manage.
 
-        Raises:
-            ValueError: if the input string is not valid.
+        Returns:
+            A boolean indicating if the passed in argument is valid.
         """
-        if not (4 <= len(s) <= 30):
-            raise ValueError(
-                ('Invalid Google Cloud Platform project name "{}": '
-                 'must be between 4 and 30 characters').format(s))
+        pass
 
 
-class DjangoProjectNamePrompt(NamePrompt):
-    """Allow the user to enter a Django project name."""
+class TemplatePrompt(Prompt):
+    """Base template for all parameter prompts interacting with the user.
 
-    _PROMPT = 'Enter a Django project name, or leave blank to use'
+    They must have a prompt method that calls one of the _x_prompt functions.
+    They must own only one parameter.
+    They should have a validate function.
+    They should call _is_valid_passed_arg at the beggining of prompt method.
+    """
 
-    @classmethod
-    def _default_name(cls, arguments: Dict[str, Any]):
-        return 'mysite'
+    # Parameter must be set for dictionary key
+    PARAMETER = None
 
-    @staticmethod
-    def validate(s, credentials: Optional[credentials.Credentials] = None):
-        """Validates that a string is a valid Django project name.
+    def prompt(self, console: io.IO, step: str,
+               args: Dict[str, Any]) -> Dict[str, Any]:
+        """Handles the business logic and calls the prompts.
 
         Args:
-            s: The string to validate.
-            credentials: The OAuth2 Credentials object to use for api calls
-                during validation.
+            console: Object to use for user I/O.
+            step: Message to present to user regarding what step they are on.
+            args: Dictionary holding prompts answered by user and set up
+                arguments.
```

Comment:
```suggestion
                command line arguments.
```

Please change everywhere.